src/document_basic.c

```diff
@@ -191,10 +194,30 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx) {
 
     // on crdt the return value might be the underline value, we must copy it!!!
     // TODO: change `fs->text` to support hash or json not RedisModuleString
-    if (japi->getJSON(json, ctx, &doc->fields[oix].text) != REDISMODULE_OK) {
-      RedisModule_Log(ctx, "verbose", "Failed to load value from field %s", field->path);
-      goto done;
-    }
+
+    if (type != JSONType_Array) {
+      if (japi->getJSON(json, ctx, &doc->fields[oix].text) != REDISMODULE_OK) {
+        RedisModule_Log(ctx, "verbose", "Failed to load value from field %s", field->path);
+        goto done;
+      }
+    } else if (field->types & INDEXFLD_T_TAG) {
+      // a tag field can index a JSON array
+      size_t arrayLen;
+      japi->getLen(json, &arrayLen);
+      RedisModuleString **ptr = doc->fields[oix].multiVal
+                              = rm_calloc(arrayLen, sizeof(RedisModuleString *));
+      doc->fields[oix].multiLen = arrayLen;
+
+      for (size_t jj = 0; jj <arrayLen; jj++) {
+        JSONType arrayJsonType;
+        RedisJSON arrayJson = japi->getAt(json, jj, &arrayJsonType);
+        if (arrayJsonType == JSONType_Array || arrayJsonType == JSONType_Object ||
+            japi->getJSON(arrayJson, ctx, &(ptr[jj])) != REDISMODULE_OK) {
```

Comment:
Why not get string? Also you need to verify that the data is string no?

Reply:
Supposedly, we can index different types of fields as other types. maybe numeric as text etc. Otherwise, you are right.
Let's bring up with @emmanuelkeller

Reply:
I think it was decided that only if all array elements are of type String, we index them
(can do it in one pass on the array elements, and if a mixture of types is found, can deallocate and clear the `multiVal` of the field, since some values could have already been loaded and we do not want to keep them)

In addition, if we know it is of type String, we can call `getString` API to get the underline raw string (with no quotes)